projects/user_publication/gscholar.py

```diff
@@ -164,28 +218,30 @@ def pub_import(
     gscholar = GoogleScholarHandler()
     pubs = []
     if not year_high:
-        year_high = datetime.now().year
+        year_high = datetime.date.today().year
     for chameleon_pub in ChameleonPublication.objects.exclude(ref__isnull=True):
         ch_pub = gscholar.get_one_pub(chameleon_pub.title)
         cited_pubs = gscholar.get_cites(ch_pub, year_low=year_low, year_high=year_high)
         for cited_pub in cited_pubs:
             authors = gscholar.get_authors(cited_pub)
             cited_pub_title = cited_pub['bib']['title']
-            if not authors:
-                continue
-            proj = utils.guess_project_for_publication(
+            project = utils.guess_project_for_publication(
                 authors, cited_pub["bib"]["pub_year"]
             )
-            if not proj:
+            if not project:
                 continue
             if ChameleonPublication.objects.filter(title__iexact=cited_pub_title).exists():
                 logger.info(f"{cited_pub_title} is a chameleon publication - ignoring")
                 continue
             # this is to find if there is a publication already in database matching to
             # this title and project however this needs to be changed after looking
             # at some examples and how to better handle these dumplicates
-            if Publication.objects.filter(title=cited_pub_title, project_id=proj).exists():
-                logger.info(f"{cited_pub_title} is already in Publications table - This check might be outdated.")
+            pub_exists = Publication.objects.filter(title=cited_pub_title, project_id=project)
+            if pub_exists:
+                add_to_all_sources(pub_exists[0], Publication.G_SCHOLAR)

```

Comment:
why is index 0 used here? I think the name `pub_exists` also should be `existing_publication` to make it more clear.

Reply:
because it is a filter, it will return a <type:queryset> so the first element is needed. Because there can be duplicates in the table now - using a `get()` will result in error